packages/core/src/session_pool/session_pool.ts

```diff
@@ -139,6 +140,8 @@ export class SessionPool extends EventEmitter {
     protected events: EventManager;
     protected readonly blockedStatusCodes: number[];
 
+    private queue = new AsyncQueue();
```

Comment:
Maybe I misunderstand the docs, but is this basically a mutex?

Reply:
I think so? Considering its basically a "hey I wanna do things" but can only do it once previous thing completes..